### PR TITLE
Fix Incorrect Time Display After Saving Transaction

### LINKED ISSUE #3424: Incorrect Time Display After Saving Transaction[BUG]
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate problems

### Describe the bug

After adding  a transaction (Income or Expence), the correct time and date are initially shown. However, once the transaction is saved and reopened, the displayed time becomes incorrect 

Here the reference 
https://github.com/user-attachments/assets/3ceec8f1-5566-4565-bf56-8a235f2a450e

### To Reproduce

1. Add an income or expense transaction in the app.
2. Note the time displayed before saving the transaction.
3. Click "Save" to save the transaction.
4. Navigate to the home page.
5. Reopen the saved transaction.
6. Observe that the time displayed has changed or is incorrect.

### Expected behavior

The expected behavior is that the time and date displayed for a transaction should remain consistent and accurate after saving and reopening the transaction. The time noted before saving should match the time shown when the transaction is reopened.

### Screenshots

_No response_

### App version

v2024.8.11 (194) from GitHub 

### Smartphone

Realme 8 Android 11

### Additional context

My Time zone : UTC+05:30 ( India )

### PR DESCRIPTION
**Pull request by shamim-emon**
## Pull request (PR) checklist

Please check if your pull request fulfills the following requirements:

- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [ ] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## What's changed?

Describe with a few bullets **what's new:**
- I've fixed  `Incorrect Time Display After Saving Transaction` issue

## Risk factors

**What may go wrong if we merge your PR?**
- No known case

**In what cases won't your code work?**
- No known case

## Does this PR close any GitHub issues? (do not delete)

- Closes #3424

## Troubleshooting GitHub Actions (CI) failures ❌
Pull request checks failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt [hunk 1]
```diff
@@ -704,7 +705,7 @@ class EditTransactionViewModel @Inject constructor(
                     dateTime = when {
                         loadedTransaction().dateTime == null &&
                                 dueDate.value == null -> {
-                            timeNowUTC()
+                            timeProvider.utcNow().atZone(timeProvider.getZoneId()).toLocalDateTime()
```
Comment (akashs056) [inline screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt:708]: I tested it, the date is showing correctly after saving the transaction but the is issue in the balance calculation. the balance didnt get updated for me...

Comment (ILIYANGERMANOV) [reply screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt:708]: Thank you @akashs056! Much appreciated that you tested 🙌

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
Folks, let's do the proper fix by reworking the entire app time handling using the `TimeConverter` API - small fixes won't solve it - they fix one thing and break another

File: screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt (hunk 1 above)
Comment (shamim-emon) [reply screen/edit-transaction/src/main/java/com/ivy/transaction/EditTransactionViewModel.kt:708]: @akashs056 thanks for pointing it out.

**Comment by shamim-emon** [comment]
> Folks, let's do the proper fix by reworking the entire app time handling using the `TimeConverter` API - small fixes won't solve it - they fix one thing and break another 

@ILIYANGERMANOV sure, since this will be a big task, can I create another issue and attach with this PR once goal is reached?

**Comment by ILIYANGERMANOV** [comment]
> > Folks, let's do the proper fix by reworking the entire app time handling using the `TimeConverter` API - small fixes won't solve it - they fix one thing and break another
> 
> @ILIYANGERMANOV sure, since this will be a big task, can I create another issue and attach with this PR once goal is reached?

Yeah that makes sense

**Comment by shamim-emon** [comment]
@ILIYANGERMANOV @akashs056 will reopen PR once done with `TimeConverter `API migration is done successfully.

**Pull request closed without merging**